### PR TITLE
Tests and documentation for systemd integration

### PR DESCRIPTION
**Pull request by vieira**
Some tests and documentation for the systemd notification feature.
Also fixes some corner case issues detected while writing the tests.

(Did I got the rst syntax right?)

### CODE REVIEW

File: docs/requirements.rst [hunk 1]
```diff
@@ -72,3 +74,19 @@ Additional Suggested Software
 
 - You may want to use autossh, available in various package management
   systems
+- If you are using systemd, sshuttle can notify it when the connection to
+  the remote end is established and the firewall rules are installed
```
Comment (brianmay) [inline docs/requirements.rst]: Should end the sentence with a full stop or colon. Yes, I know, looks line line 76 is wrong too.

File: docs/requirements.rst [hunk 2]
```diff
@@ -72,3 +74,19 @@ Additional Suggested Software
...
+
+.. code-block:: ini
+   :emphasize-lines: 6
```
Comment (brianmay) [inline docs/requirements.rst]: Why do you want to emphasise line 6? Looks like that is the line with ``Type=notify``.

File: docs/requirements.rst (hunk 1 above)
Comment (vieira) [inline docs/requirements.rst]: Thanks for the feedback, pushed a new commit adding full stops to both lines.

File: docs/requirements.rst (hunk 2 above)
Comment (vieira) [inline docs/requirements.rst]: Pushed a new commit explaining that the the process start-up type for sshuttle service unit should be set to notify for this feature to work. Thanks for reviewing @brianmay. Let me know if you have any other observations. Thanks!
